examples/main.rs.tpl

```diff
@@ -0,0 +1,135 @@
+    let action_tree = ActionTree::new(tree_config).unwrap();
+    let mut game = PostFlopGame::with_config(card_config, action_tree).unwrap();
+
+    // obtain the private hands
+    let oop_hands = game.private_cards(0);
+    assert_eq!(
+        holes_to_string(&oop_hands[..10]).unwrap(),
+        "5c4c,Ac4c,5d4d,Ad4d,5h4h,Ah4h,5s4s,As4s,6c5c,7c5c"
+    );
+
+    // check memory usage
+    let (mem_usage, mem_usage_compressed) = game.memory_usage();
+    println!(
+        "Memory usage without compression: {:.2}GB",
+        mem_usage as f64 / (1024.0 * 1024.0 * 1024.0)
+    );
+    println!(
+        "Memory usage with compression: {:.2}GB",
+        mem_usage_compressed as f64 / (1024.0 * 1024.0 * 1024.0)
+    );
+
+    // allocate memory without compression
+    game.allocate_memory(false);
+
+    // allocate memory with compression
+    // game.allocate_memory(true);
+
+    // solve the game
+    let max_num_iterations = 1000;
+    let target_exploitability = game.tree_config().starting_pot as f32 * 0.005;
+    let exploitability = solve(&mut game, max_num_iterations, target_exploitability, true);
+    println!("Exploitability: {:.2}", exploitability);
+
+    // solve the game manually
+    // for i in 0..max_num_iterations {
+    //     solve_step(&game, i);
+    //     if (i + 1) % 10 == 0 {
+    //         let exploitability = compute_exploitability(&game);
+    //         if exploitability <= target_exploitability {
+    //             println!("Exploitability: {:.2}", exploitability);
+    //             break;
+    //         }
+    //     }
+    // }
+    // finalize(&mut game);
+
+    // get equity and EV of a specific hand
+    game.cache_normalized_weights();
+    let equity = game.equity(0); // `0` means OOP player
+    let ev = game.expected_values(0);
+    println!("Equity of oop_hands[0]: {:.2}%", 100.0 * equity[0]);
+    println!("EV of oop_hands[0]: {:.2}", ev[0]);
+
+    // get equity and EV of whole hand
+    let weights = game.normalized_weights(0);
+    let average_equity = compute_average(&equity, weights);
+    let average_ev = compute_average(&ev, weights);
+    println!("Average equity: {:.2}%", 100.0 * average_equity);
+    println!("Average EV: {:.2}", average_ev);
+
+    // get available actions (OOP)
+    let actions = game.available_actions();
+    assert_eq!(
+        format!("{:?}", actions),
+        "[Check, Bet(120), Bet(216), AllIn(900)]"
+    );
+
+    // play `Bet(120)`
+    game.play(1);
+
+    // get available actions (IP)
+    let actions = game.available_actions();
+    assert_eq!(format!("{:?}", actions), "[Fold, Call, Raise(300)]");
+
+    // confirm that IP does not fold the nut straight
+    let ip_cards = game.private_cards(1);
+    let strategy = game.strategy();
+    assert_eq!(ip_cards.len(), 250);
+    assert_eq!(strategy.len(), 750);
+    assert_eq!(hole_to_string(ip_cards[206]).unwrap(), "KsJs");
+    assert_eq!(strategy[206] + strategy[456] + strategy[706], 1.0);
```

Comment:
This seems to be true now, but dealing with floating point numbers can be tricky. We should check this by:
```suggestion
    assert!((strategy[206] + strategy[456] + strategy[706] - 1.0).abs() < 1e-6);
```